lib/widgets.php

```diff
@@ -181,3 +181,88 @@ function gutenberg_create_wp_area_post_type() {
 	);
 }
 add_action( 'init', 'gutenberg_create_wp_area_post_type' );
+
+/**
+ * Outputs a block widget on the website frontend.
+ *
+ * @param array $options   Widget options.
+ * @param array $arguments Arguments array.
+ */
+function gutenberg_output_blocks_widget( $options, $arguments ) {
+	echo $options['before_widget'];
+	foreach( $arguments['blocks'] as $block ) {
+		echo render_block( $block );
+	}
+	echo $options['after_widget'];
+}
+
+/**
+ * Registers of a widget that should represent a set of blocks and returns its id.
+ *
+ * @param array $blocks   Array of blocks.
+ */
+function gutenberg_blocks_to_widget( $blocks ) {
+	$widget_id = 'blocks-widget-' . md5( Experimental_WP_Widget_Blocks_Manager::serialize_block( $blocks ) );
```

Comment:
Let's pull this function out of `Experimental_WP_Widget_Blocks_Manager`. We already have `serialize_blocks` so it makes sense in my mind that we'd have `serialize_block` as well.

Reply:
Hi @noisysocks, in other PR @youknowriad recommended that we keep serialize_blocks inside Experimental_WP_Widget_Blocks_Manager to reinforce the experimental nature of the function. Given that recomendation, I guess we should also keep function gutenberg_blocks_to_widget( $blocks ) inside.

Reply:
Also what does serializing means? Is it's the same as the frontend? If it is, it means the blocks should be fully parsed and not just half-parsed.

Reply:
I think the serialization we have here is very simple and may not even work on all cases yet. It only needs to work for legacy widgets. I guess it is better to keep it inside the experimental class. And then we may have to rethink this to simply have functions like serialize_legacy_widgets which make their objectives more clear and may end up being public.